packages/api/src/utils/db-client.js

```diff
@@ -554,19 +563,62 @@ export class DBClient {
         'uploads_multipart_total',
       ])
 
-    if (error) {
-      throw new DBError(error)
+    const dagByteSizeQuery = metricsQuery
+      .select('name, dimensions, value')
+      .match({
+        name: 'dagcargo_project_bytes_in_active_deals',
+        dimensions: '{{project,nft.storage}}',
+      })
+      .single()
+
+    const weekAgo = new Date()
+    weekAgo.setDate(weekAgo.getDate() - 7)
+
+    const dagByteSizeHistory = metricsLogQuery
+      .select('name, dimensions, value')
+      .match({
+        name: 'dagcargo_project_bytes_in_active_deals',
+        dimensions: '{{project,nft.storage}}',
+      })
+      .lte('collected_at', weekAgo.toISOString())
+      .order('collected_at', { ascending: false })
+      .range(0, 1)
+      .single()
+
+    const [primaryRes, dagSizeRes, dagSizeHistRes] = await Promise.all([
+      primaryMetricQuery,
+      dagByteSizeQuery,
+      dagByteSizeHistory,
+    ]).catch((err) => {
+      throw new Error(err)
+    })
+
+    const { data: primaryMetrics, error: primaryMetricsError } = primaryRes
+
+    const { data: dealsSize, error: dealsSizeError } = dagSizeRes
+
+    const { data: dealsSizeHistory, error: dealsSizeHistoryError } =
+      dagSizeHistRes
+
+    if (primaryMetricsError || dealsSizeHistoryError || dealsSizeError) {
+      // @ts-ignore
+      throw new DBError('Error fetching stats.')
     }
 
-    if (!data || !data.length) {
-      return undefined
+    /** @type any  */
+    const outboundData = {}
```

Comment:
IDK maybe `stats` is a better name for this?

```suggestion
    const stats = {}
```

Reply:
done